module/zfs/dsl_scan.c

```diff
@@ -611,11 +612,13 @@ dsl_scan_prefetch(dsl_scan_t *scn, arc_buf_t *buf, blkptr_t *bp,
 	    (BP_GET_LEVEL(bp) == 0 && BP_GET_TYPE(bp) != DMU_OT_DNODE))
 		return;
 
+	if (BP_IS_ENCRYPTED(bp))
+		zio_flags |= ZIO_FLAG_RAW;
```

Comment:
> Since this is the prefetch i/o, are you sure RAW would cause a problem?

It probably would not have much of an impact, if any at all, but I didn't want to make unnecessary changes to code that was already working. Theoretically, it could have an impact for instances that have turned off compressed arc, but would otherwise be a noop.

> The check on 612 still lets through non-dnode (and non-encrypted) blocks - specifically indirect blocks. If it would be bad to use RAW on indirect blocks, we should have an assertion here.

Indirect blocks are never encrypted, so specifying `RAW` here would  effectively be the same as `RAW_COMPRESS`. For prefetch, the arc will do that anyway if compressed arc is enabled. If compressed arc is not enabled that might cause problems. I can add an assert.